pkg/cli/search_cmd.go

```diff
@@ -0,0 +1,148 @@
+//go:build search
+// +build search
+
+package cli
+
+import (
+	"os"
+	"path"
+
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+
+	zotErrors "zotregistry.io/zot/errors"
+)
+
+//nolint:dupl
+func NewSearchCommand(searchService SearchService) *cobra.Command {
+	searchImageParams := make(map[string]*string)
+
+	var servURL, user, outputFormat string
+
+	var isSpinner, verifyTLS, verbose, debug bool
+
+	imageCmd := &cobra.Command{
+		Use:   "search [config-name]",
+		Short: "Search images and their tags",
+		Long: `Search repos or images
+Example:
+  # For repo search specify a substring of the repo name without the tag
+  zli search --query test/repo
+
+  # For image search specify the full repo name followed by the tag or a prefix of the tag.
+  zli search --query test/repo:2.1.
+		`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			home, err := os.UserHomeDir()
+			if err != nil {
+				panic(err)
+			}
+
+			configPath := path.Join(home + "/.zot")
+			if servURL == "" {
+				if len(args) > 0 {
+					urlFromConfig, err := getConfigValue(configPath, args[0], "url")
+					if err != nil {
+						cmd.SilenceUsage = true
+
+						return err
+					}
+
+					if urlFromConfig == "" {
+						return zotErrors.ErrNoURLProvided
+					}
+
+					servURL = urlFromConfig
+				} else {
+					return zotErrors.ErrNoURLProvided
+				}
+			}
+
+			if len(args) > 0 {
+				var err error
+				isSpinner, err = parseBooleanConfig(configPath, args[0], showspinnerConfig)
+				if err != nil {
+					cmd.SilenceUsage = true
+
+					return err
+				}
+
+				verifyTLS, err = parseBooleanConfig(configPath, args[0], verifyTLSConfig)
+				if err != nil {
+					cmd.SilenceUsage = true
+
+					return err
+				}
+			}
+
+			spin := spinner.New(spinner.CharSets[39], spinnerDuration, spinner.WithWriter(cmd.ErrOrStderr()))
+			spin.Prefix = prefix
+
+			searchConfig := searchConfig{
+				params:        searchImageParams,
+				searchService: searchService,
+				servURL:       &servURL,
+				user:          &user,
+				outputFormat:  &outputFormat,
+				verbose:       &verbose,
+				debug:         &debug,
+				spinner:       spinnerState{spin, isSpinner},
+				verifyTLS:     &verifyTLS,
+				resultWriter:  cmd.OutOrStdout(),
+			}
+
+			err = globalSearch(searchConfig)
+
+			if err != nil {
+				cmd.SilenceUsage = true
+
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	setupSearchFlags(imageCmd, searchImageParams, &servURL, &user, &outputFormat, &verbose, &debug)
+	imageCmd.SetUsageTemplate(imageCmd.UsageTemplate() + usageFooter)
+
+	return imageCmd
+}
+
+func setupSearchFlags(imageCmd *cobra.Command, searchImageParams map[string]*string,
+	servURL, user, outputFormat *string, verbose *bool, debug *bool,
+) {
+	searchImageParams["query"] = imageCmd.Flags().StringP("query", "q", "",
+		"Specify what repo or image(repo:tag) to be searched")
+
+	searchImageParams["subject"] = imageCmd.Flags().StringP("subject", "s", "", "List all referrers for this subject")
```

Comment:
Mention the format in which to specify the subject.

Reply:
Pls document more about this.